src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,60 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String SUPPLY_OPERATION = "supply";
+    private static final String BUY_OPERATION = "buy";
+
+    public static void getStatistic(String fromFileName, String toFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
+                BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+
+            int supplyTotal = 0;
+            int buyTotal = 0;
+
+            String line;
+            while ((line = readFile(reader)) != null) {
+                String[] fields = line.split(",");
+
+                if (fields.length == 2) {
+                    String operationType = fields[0].trim();
+                    int amount = Integer.parseInt(fields[1].trim());
+
+                    if (SUPPLY_OPERATION.equals(operationType)) {
+                        supplyTotal += amount;
+                    } else if (BUY_OPERATION.equals(operationType)) {
+                        buyTotal += amount;
+                    }
+                }
+            }
+
+            String report = formReport(supplyTotal, buyTotal);
+            writeToFile(writer, report);
+
+        } catch (IOException e) {
+            handleException(e, fromFileName);
+        }
+    }
```

Comment:
feel free to use this template. at the moment, your method is too big and cares more than one resonsibility, which violates SRP
```suggestion
    public static void getStatistic(String fromFileName, String toFileName) {
        String dataFromFile = readFile(fromFileName);
        String report = createReport(dataFromFile);
        writeReportToFile(report, toFileName);
    }
```